### DISCUSSION TITLE
Unable to install Stylix with my configuration

### DISCUSSION THREAD

**Discussion by nonetrix** [nix-community/stylix#199] [Q&A] [answered]
I have tried to add Stylix by changing make flake.nix to this:
```
# flake.nix
{
  inputs = {
    nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
    home-manager = {
      url = "github:nix-community/home-manager";
      #inputs.nixpkgs.follows = "nixpkgs";
    };

    hyprland.url = "github:hyprwm/Hyprland";

    #This should be default change my mind
    split-monitor-workspaces = {
      url = "github:Duckonaut/split-monitor-workspaces";
      inputs.hyprland.follows = "hyprland"; # <- make sure this line is present for the plugin to work as intended
    };

    stylix.url = "github:danth/stylix";
  };

  outputs = {
    self,
    nixpkgs,
    home-manager,
    stylix,
    ...
  } @ inputs: let
    system = "x86_64-linux";

    pkgs = import nixpkgs {
      inherit system;

      config = {
        allowUnfree = true;
      };
    };
  in {
    nixosConfigurations = {
      nixos = nixpkgs.lib.nixosSystem {
        specialArgs = {inherit inputs stylix;};
        system = "x86_64-linux";
        modules = [
          ./configuration.nix
          home-manager.nixosModules.home-manager
          stylix.homeManagerModules.stylix
          {
            home-manager = {
              useGlobalPkgs = true;
              useUserPackages = true;
            };
          }
        ];
      };
    };
  };
}
```
However, whenever I run `sudo nixos-rebuild switch --flake .` it returns the following:
```
warning: Git tree '/home/noah/Documents/Projects/nixdots' is dirty
error:
       … while calling the 'seq' builtin

         at /nix/store/nn17kpb3mlp3kdhi4vm1m71w0m68f1gk-source/lib/modules.nix:320:18:

          319|         options = checked options;
          320|         config = checked (removeAttrs config [ "_module" ]);
             |                  ^
          321|         _module = checked (config._module);

       … while calling the 'throw' builtin

         at /nix/store/nn17kpb3mlp3kdhi4vm1m71w0m68f1gk-source/lib/modules.nix:296:18:

          295|                     ''
          296|             else throw baseMsg
             |                  ^
          297|         else null;

       error: The option `dconf' does not exist. Definition values:
       - In `/nix/store/3znkj7gxai15wcwr16sxwmkn340gqrb7-modules/gnome/hm.nix'
 ```
I have no idea why or if it is a issue on my end or not, I suspect I am doing something wrong perhaps. Anyway, here is my configs, be sure to change branch to the Stylix branch as Master doesn't have any changes. I just want to be pointed in the right direction or told what I did wrong as I read the documentation and also followed examples: https://github.com/nonetrix/nixdots 

It seems to be something to do with this file and this line: https://github.com/danth/stylix/blob/b5ad31b710294038f9ed70efdf787db6a82d7327/modules/gnome/hm.nix#L10

Dconf seems to be like the Windows Registry but for GNOME and GTK apps etc. So logically I looked in home manager code to see if I could find it, and sure enough it's here and has been here for a while so judging off commit history. So it's not like my home manager is too old or anything and doesn't have the module, also it's not like it's too new probably because it hasn't been updated recently. https://github.com/nix-community/home-manager/blob/master/modules/misc/dconf.nix

I will admit I am fairly newbie at Nix and Flakes and etc. Most of my knowledge has been gained from using it 9 months ago and naturally learning it, then switching back to Arch for a while, then now where I have switched back to it and revamped my configs completely to use Hyprland, Flakes, and to use my AMD GPU as opposed to NVIDIA etc.

**Comment by danth** [comment, marked as answer]
In your modules list you should be importing `stylix.nixosModules.stylix` instead of `stylix.homeManagerModules.stylix` since it's a NixOS configuration (as opposed to an independent Home Manager configuration).

Let me know if that fixes your issue :)

**Reply by nonetrix** [reply]
Works now, can't get the automatic wallpaper setting to work on Hyprland though, and QT theming doesn't look right but https://github.com/danth/stylix/pull/142 seems to be a step in the right direction

**Reply by danth** [reply]
Hyprland wallpapers are coming in #102
